Sources/Java2SwiftLib/JavaTranslator.swift

```diff
@@ -458,6 +472,31 @@ extension JavaTranslator {
       """
   }
 
+  package func translateToEnumValue(name: String, enumFields: [Field]) -> [DeclSyntax] {
+    let extensionSyntax: DeclSyntax = """
+      public enum \(raw: name): Equatable {
+        \(raw: enumFields.map { "case \($0.getName())" }.joined(separator: "\n"))
+      }
+    """
+
+    let mappingSyntax: DeclSyntax = """
+      public var enumValue: \(raw: name)? {
+        \(raw: enumFields.map {
+          // The equals method takes a java object, so we need to cast it here
+          """
+          if self.equals(self.javaClass.\($0.getName())?.as(JavaObject.self)) {
```

Comment:
Could you put `self.javaClass` into a local variable so we aren't accessing this computed property every time?